nipype/pipeline/engine/nodes.py

```diff
@@ -627,7 +627,8 @@ def _run_command(self, execute, copyfiles=True):
             self._interface.__class__.__name__)
         if issubclass(self._interface.__class__, CommandLine):
             try:
-                cmd = self._interface.cmdline
+                with indirectory(outdir):
```

Comment:
does outdir always exist at this point?

Reply:
We create the `command.txt` file in line 637, just below here, and copy files to working directory just above.

A quick scan didn't show me where the directory is created, but I'm reasonable confident it exists.

Reply:
The directory is created here, inside `run()`:

https://github.com/effigies/nipype/blob/555a905f623be290a16c5fede1ea3a1558ac1f57/nipype/pipeline/engine/nodes.py#L469

Therefore, it is not granted that it will exist at this point. We could add a `makedirs` call to the context manager.

Reply:
but you can always call cmdline manually before run.

inside a workflow this does not apply, but many people use the interfaces and call cmdline in their ipython interactive sessions.